idaes/core/util/unit_costing.py

```diff
@@ -33,15 +35,37 @@ def global_costing_parameters(self, year=None):
     self.CE_index = Param(mutable=True, initialize=ce_index_dic[year],
                           doc='Chemical Engineering Plant Cost Index $ year')
 
+    if integer_n_units not in [True, False]:
+        raise ValueError('integer_n_units must be True or False')
```

Comment:
I would suggest making this the following:

`"{} - integer_n_units must be True or False".format(self.name)`

That way, the user gets some information about which model/block has the issue without needing to try to follow the traceback.